### PR TITLE
Merge my personal messages

### PR DESCRIPTION
**Pull request by violine1101**
Since I plan on no longer using my personal database of messages, I added all messages that were not present in this repo yet and that I thought might be useful here.

Also includes some more improvements on existing messages. Please report back if you have any issues with what I changed.

Additionally, I have moved over some more commonly used things into variables, and compacted the existing variables down a bit. This also means that ⚠ **_this is a breaking change_** ⚠ since variables can include other variables now, which is also why I needed to adjust the JavaScript file a bit (recursion limit: 10). Other projects that rely on this may not expect that and might not be able to parse variables containing other variables correctly.

This PR is based on #61, so that should be merged before this one. I made a separate PR because this is a much larger change and is more likely to take longer to discuss.

### CODE REVIEW

**Comment by SPGoding** [comment]
As this is already a breaking changes, how do you think about moving the new `shortcut`s to the keys? I don't feel like we need two different strings that can both be used to locate a certain message

File: assets/js/index.js [hunk 1]
```diff
@@ -79,15 +79,30 @@ $("#copybutton").click(function () {
...
       }
     }
+
+    if (varExpanded && limit > 0) {
+      return resolveVariables(text, limit - 1);
+    } else {
+      return text;
+    }
   }
-  // TODO: Allow more than one variable
```
Comment (SPGoding) [inline assets/js/index.js:90]: (just a small note: the comment here is actually for the `message.fillname` part)

Comment (violine1101) [reply assets/js/index.js:90]: Ah, whoops, i thought it was about what I implemented.
Anyway, if that's really needed, it should probably be an issue at this point, but I don't really see the need for multiple inputs as of right now.

**Comment by violine1101** [comment]
That's true. I wasn't sure if the keys were used for anything significant anywhere, so I kept them as it was.

Arisa is prepared now for this change, so it shouldn't be as breaking as it was anymore, it was a easier fix than I thought. Unless I've forgotten something else that relies on the helper messages, we should be good now.

If we go for breaking changes, it might make sense to revamp the format of the file completely. For instance on many messages you might just want to specify "all" projects. That's currently not possible. That could be fixed by for instance leaving out the "projects" property if all projects should be specified. And `fillname` must always be specified too, that's pretty annoying. But those things should probably be discussed in another issue.

File: assets/messages.yml [hunk 2]
```diff
@@ -761,15 +928,32 @@ messages:
...
-        Minecraft has several different editions, and some features vary from edition to edition. Often this happens when one edition evolves faster than another, and tweaks or balances are made that don't propagate immediately to other editions.
-        Parity issues are not tracked on the bug tracker but should instead be reported on the [Feedback website|https://feedback.minecraft.net/hc/en-us/community/topics/360001191251-Parity].
+        *Thank you for your report!*
+        However, this issue is {color:#FF5722}*Invalid*{color}.
+
+        You have posted a parity issue. Minecraft has several different editions, and some features vary from edition to edition.
+        This is especially the case with features that were added a long time ago, because one edition evolved faster than another, and tweaks or balances were made that did not propagate immediately to other editions.
+        These differences between the different versions of the game are called "Parity issues".
+
+        However, we only accept bug reports about a parity issue if it fulfills the following criteria:
```
Comment (Marcono1234) [inline assets/messages.yml:940]:
```suggestion
        We only accept bug reports about a parity issue if it fulfills the following criteria:
```
This does not conflict with the previous paragraph so "However" seems to not be necessary.

File: assets/messages.yml [hunk 3]
```diff
@@ -902,13 +1102,14 @@ messages:
...
         *Thank you for your report!*
         However, this issue is {color:#FF5722}*Working as Intended*{color}.
 
         The report you have submitted is working as intended: %s%.
 
-        Please note, that mechanics of the game may change between updates.
+        Please note that mechanics of the game may change between updates.
         Things such as graphics, sounds, world creation, biomes, redstone, villagers, and animals may not work the same in current versions.
 
         [Full Version History|https://minecraft.gamepedia.com/Java_Edition_version_history] -- [Snapshot Version History|https://minecraft.gamepedia.com/Java_Edition_version_history/Development_versions] -- [Feature Requests and Suggestions|https://www.reddit.com/r/minecraftsuggestions]
```
Comment (Marcono1234) [inline assets/messages.yml:1115]: The "Feature Requests and Suggestions" link here (same for mcpe and mcl) is redundant because it is in the quicklinks as well (though there it points to the feedback site).
Should it be removed?

File: assets/messages.yml [hunk 4]
```diff
@@ -351,10 +416,36 @@ messages:
...
+        - mctest
+        - realms
+        - web
+      name: Duplicate (resolved)
+      shortcut: rdup
+      message: |-
+        *Thank you for your report!*
+        We're tracking this issue as *%s%*, so this ticket is being resolved and linked as a *duplicate*.
+
+        That ticket has already been resolved, which means this is not considered a bug. Please take a look at that ticket to learn more.
```
[the remark this answers is not in the document]
Comment (violine1101) [reply assets/messages.yml:445]: I was thinking of Invalid tickets in specific. You're right with "this is not considered a bug" it doesn't work for Won't fix, which is another case where I would have liked to use that message.

File: assets/messages.yml [hunk 5]
```diff
@@ -173,36 +226,43 @@ messages:
...
         %quick_links%
       fillname: []
-  crash-logges-automatically:
+  crash-logged-automatically:
     - project:
         - bds
         - mcpe
         - realms
       name: Crash logged automatically
+      shortcut: crash
```
[the remark this answers is not in the document]
Comment (violine1101) [reply assets/messages.yml:265]: The projects of the separate `crash` shortcuts are mutually exclusive, so it's not an issue. But if that changes, it would become one. I mostly chose `crash` because I couldn't think of any better shortcut.

The messages are all sorted alphabetically. But perhaps it would make sense to merge them into a single `crash` message for it to be less confusing.

File: assets/messages.yml [hunk 6]
```diff
@@ -446,38 +555,45 @@ messages:
...
         - mc
-        - mcapi
-        - mcd
-        - mce
         - mcl
-        - mcpe
+      name: Attach hserr_pid file
+      shortcut: hserr
+      message: |-
+        Is there a file with a name starting with {{hs_err_pid}} in your {{.minecraft}} folder? If so, can you attach that here?
```
[the remark this answers is not in the document]
Comment (violine1101) [reply assets/messages.yml:583]: Good call, it probably should have that link. I used to always type that message manually so I didn't include it.

File: assets/messages.yml [hunk 7]
```diff
@@ -601,17 +740,35 @@ messages:
...
         Please put only one bug report in each ticket. It is very difficult to keep track of bugs when they are not in their own tickets, and it is easier for you to make multiple tickets than for the mods to separate them.
 
+        %search_for_dupes%
+
+        %quick_links%
+      fillname: []
+  non-api-feature-request:
+    - project:
+        - mcapi
+      name: Non-API feature request
```
[the remark this answers is not in the document]
Comment (violine1101) [reply assets/messages.yml:757]: No, it's not really, but there are still occasionally new tickets coming in. I've seen gameplay feature requests on there, that's what this message is for.

File: assets/messages.yml (hunk 3 above)
Comment (violine1101) [reply assets/messages.yml:1115]: I'd say that probably all of these links should be removed, they all don't really seem too relevant to me. But personally I don't use this message anyway.

The link to minecraftsuggestions is there because the link to the feedback site was only added a lot later.

File: assets/messages.yml [hunk 8]
```diff
@@ -220,11 +281,23 @@ messages:
...
         If you need help or have a suggestion you might like to follow a link below.
 
+        %quick_links%
+      fillname: []
+    - project: mcd
+      name: Crash logged automatically
+      shortcut: crash
+      message: |-
+        *Thank you for your report!*
+        However, this issue is {color:#FF5722}*Invalid*{color}.
```
[the remark this answers is not in the document]
Comment (violine1101) [reply assets/messages.yml:291]: I don't think it is. But I don't really want to tamper too much with the MCPE messages, I don't really know how they're used.

File: assets/messages.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (violine1101) [reply assets/messages.yml]: Changed it to `autocrash`, just to be more explicit.

File: assets/messages.yml [hunk 9]
```diff
@@ -117,6 +144,21 @@ messages:
...
         Could you please include information about your server? Without it we may be unable to reproduce the issue.
 
+        %quick_links%
+      fillname: []
+  attach-launcher-log:
+    - project:
+        - mc
+        - mcl
+      name: Attach launcher log
+      shortcut: llog
```
Comment (NeunEinser) [inline assets/messages.yml:154]: Why not simply log? There doesn't seem to be any message that currently uses log anyways.

File: assets/messages.yml [hunk 10]
```diff
@@ -180,6 +224,24 @@ messages:
...
+        - mctest
+        - realms
+        - web
+      name: Awaiting Response
+      shortcut: ar
+      message: |-
+        %awaiting_response%
+
         %quick_links%
       fillname: []
```
Comment (NeunEinser) [inline assets/messages.yml:246]: I'm not entirely sure how this is intended to be used, but maybe it would make sense to add a variable to this one for the question?

File: assets/messages.yml [hunk 11]
```diff
@@ -308,53 +353,53 @@ messages:
...
-        -- If that did not help, please contact the *[Community Support|https://discord.gg/58Sxm23]*.
+        -- If that did not help, please contact *[Community Support|https://discord.gg/58Sxm23]*.
 
         %quick_links%
       fillname: []
   duplicate-of-mcl-5638:
     - project: mcl
-      name: Duplicate of MCL-5638
-      shortcut: jre
+      name: MCL-5638 / Unable to locate Java runtime
```
Comment (NeunEinser) [inline assets/messages.yml:395]: I did like those being sorted in the box together (because all duplicate messages used to start with "Duplicate".

But tbh, the dropdown becomes more and more crowded anyways, making it harder to find stuff, and we should probably think about a better solution in general.

File: assets/messages.yml [hunk 12]
```diff
@@ -376,6 +421,31 @@ messages:
...
+        *Thank you for your report!*
+        We're tracking this issue as *%s%*, so this ticket is being resolved and linked as a *duplicate*.
+
+        That ticket has already been resolved. Please take a look at it to learn more.
+
+        %search_for_dupes%
+
         %quick_links%
       fillname:
         - Enter parent issue ID
```
Comment (NeunEinser) [inline assets/messages.yml:451]: Why do we need that message? We already have individual messages for fixed, tech support, wai and won't fix.

Other resolutions are:
* Incomplete, but we don't use Incomplete tickets as a parent (or otherwise reopen it)
* Cannot Reproduce, but that either means it's fixed, or it should be reopened
* AR - might make sense to add a message for that saying something like "... but the parent is currently missing vital information. Please do take a look and see if you can help out, in order to get this bug fixed as soon as possible"
* Done - is not used anymore afaik.

File: assets/messages.yml [hunk 13]
```diff
@@ -826,8 +952,23 @@ messages:
...
+        * The feature affected by the parity issue is present in both Bedrock Edition and Java Edition in the latest release or development version
+        * The feature behaves differently in one edition than in the other
+        * The parity issue was introduced in Buzzy Bees (Bedrock Edition 1.14 / Java Edition 1.15) or later and was not present before
+
+        All other parity issues are not tracked on the bug tracker but should instead be reported on the [Feedback website|https://feedback.minecraft.net/hc/en-us/community/topics/360001191251-Parity].
+
+        For more information about this, please take a look at [this post|https://www.reddit.com/r/Mojira/comments/g9rjfh/from_now_on_well_accept_bug_reports_about_certain/] on our subreddit.
+
+        %quick_links%
       fillname: []
```
Comment (NeunEinser) [inline assets/messages.yml:972]: That message has been changed already, is this change still required / better than the one we already have?

File: assets/messages.yml (hunk 3 above)
Comment (NeunEinser) [reply assets/messages.yml:1139]: I think it would be better to change / discuss this in a seperate issue / pr.

File: assets/messages.yml (hunk 9 above)
Comment (Marcono1234) [reply assets/messages.yml:154]: Though `log` could be used for the game log in the future

File: assets/messages.yml (hunk 12 above)
Comment (Marcono1234) [reply assets/messages.yml:451]: In theory we are missing one for "Invalid", though this message might not be that useful for this either?

File: assets/messages.yml (hunk 3 above)
Comment (Marcono1234) [reply assets/messages.yml:1139]: Created #87 for this

File: assets/messages.yml (hunk 9 above)
Comment (NeunEinser) [reply assets/messages.yml:154]: The game log is included in "Attach crash report".

File: assets/messages.yml (hunk 12 above)
Comment (NeunEinser) [reply assets/messages.yml:451]: Well, we have one for tech support. Account issues obv. don't get linked, and a duplicate feature request doesn't happen often, and tbh I don't bother searching for a parent in this case and just resolve as invalid.

File: assets/messages.yml [hunk 14]
```diff
@@ -489,23 +573,27 @@ messages:
...
         - mc
-        - mcapi
-        - mcd
-        - mce
         - mcl
-        - mcpe
+      name: Attach hserr_pid file
+      shortcut: hserr
+      message: |-
+        Is there a file with a name starting with {{hs_err_pid}} in your {{[https://minecrafthopper.net/help/guides/finding-minecraft-data-folder/|.minecraft]}} folder? If so, please attach it here.
```
Comment (Marcono1234) [inline assets/messages.yml:583]: We should also tell the user to make their report private, see [MC-207889](https://bugs.mojang.com/browse/MC-207889)

**Comment by violine1101** [comment]
This has diverged too much from the main branch and I also believe that the current messages are okay. I might still pick one or two of these changes up later but that would be separate, smaller PRs.
